compiler/extccomp.nim

```diff
@@ -493,22 +493,25 @@ proc needsExeExt(conf: ConfigRef): bool {.inline.} =
            (conf.target.hostOS == osWindows)
 
 proc useCpp(conf: ConfigRef; cfile: AbsoluteFile): bool =
-  conf.backend == backendCpp and not cfile.string.endsWith(".c")
+  # List of possible file extensions taken from gcc
+  for ext in [".C", ".cc", ".cpp", ".CPP", ".c++", ".cp", ".cxx"]:
+    if cfile.string.endsWith(ext): return true
+  false
```

Comment:
What was wrong with the old logic? It was faster... :P

Reply:
it matches objective c files (`.m`) which breaks mac

Reply:
ditto `.asm` and similar - there are two main problems with matching the wrong language:
* flags like `-std=gnu++14` are not supported
* linking with libstdc++ is costly and needs only happen if some c++ was processed

Reply:
Ok.